### PR TITLE
[4189] [studio] Publisher doesn't work when a site with a published repo is removed and then a new site is created with same site name.

### PR DESCRIPTION
**Pull request by dejan-brkic**
Fixed issue caused by side-effect of lazy published repo
Build issues
Code quality issues

### Ticket reference or full description of what's in the PR
https://github.com/craftercms/craftercms/issues/4189

### CODE REVIEW

File: src/main/java/org/craftercms/studio/api/v1/asset/Asset.java [hunk 1]
```diff
@@ -32,16 +32,10 @@ public Asset(String repoPath, Path filePath) {
         this.filePath = filePath;
     }
 
-    /**
-     * Returns the repo path for the asset.
-     */
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/api/v1/asset/Asset.java:37]: Why remove comments?

File: src/main/java/org/craftercms/studio/api/v1/asset/processing/ProcessorConfiguration.java [hunk 2]
```diff
@@ -28,30 +28,18 @@
...
     }
 
-    /**
-     * Sets the processor type.
-     */
     public void setType(String type) {
         this.type = type;
     }
 
-    /**
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/api/v1/asset/processing/ProcessorConfiguration.java:45]: Why remove comments?

File: src/main/java/org/craftercms/studio/api/v1/asset/processing/ProcessorPipelineConfiguration.java [hunk 3]
```diff
@@ -28,44 +28,26 @@
     private boolean keepOriginal;
     private List<ProcessorConfiguration> processorsConfig;
 
-    /**
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/api/v1/asset/processing/ProcessorPipelineConfiguration.java:31]: Why remove comments?

File: src/main/java/org/craftercms/studio/api/v1/aws/elastictranscoder/TranscoderJob.java [hunk 4]
```diff
@@ -27,44 +27,26 @@
     private String outputBucket;
     private String baseKey;
 
-    /**
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/api/v1/aws/elastictranscoder/TranscoderJob.java:30]: Why remove comments?

File: src/main/java/org/craftercms/studio/api/v1/aws/elastictranscoder/TranscoderOutput.java [hunk 5]
```diff
@@ -27,29 +27,27 @@
...
     private String thumbnailSuffixFormat;
 
-    /**
-     * Returns the ID of the preset with the transcoding configuration.
-     */
     public String getPresetId() {
         return presetId;
     }
 
-    /**
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/api/v1/aws/elastictranscoder/TranscoderOutput.java:37]: Why remove comments?

File: src/main/java/org/craftercms/studio/api/v1/aws/elastictranscoder/TranscoderProfile.java [hunk 6]
```diff
@@ -29,31 +29,19 @@
 
     private String pipelineId;
     private List<TranscoderOutput> outputs;
-
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/api/v1/aws/elastictranscoder/TranscoderProfile.java:32]: Why remove comments?

File: src/test/java/org/craftercms/studio/impl/v2/repository/blob/BlobAwareContentRepositoryTest.java [hunk 7]
```diff
@@ -25,7 +25,7 @@
 import org.craftercms.studio.api.v1.to.VersionTO;
 import org.craftercms.studio.api.v2.repository.blob.StudioBlobStore;
 import org.craftercms.studio.api.v2.repository.blob.StudioBlobStoreResolver;
-import org.craftercms.studio.impl.v1.repository.git.GitContentRepository;
+import org.craftercms.studio.impl.v2.repository.GitContentRepository;
```
Comment (sumerjabri) [inline src/test/java/org/craftercms/studio/impl/v2/repository/blob/BlobAwareContentRepositoryTest.java:28]: This breaks this test.

**Review by sumerjabri — CHANGES_REQUESTED** [review]

File: src/main/java/org/craftercms/studio/api/v1/asset/Asset.java (hunk 1 above)
Comment (dejan-brkic) [reply src/main/java/org/craftercms/studio/api/v1/asset/Asset.java:37]: There is a recommendation not to document setters/getters and/or obvious stuff in order not to generate redundant or useless docs

File: src/main/java/org/craftercms/studio/api/v1/asset/processing/ProcessorConfiguration.java (hunk 2 above)
Comment (dejan-brkic) [reply src/main/java/org/craftercms/studio/api/v1/asset/processing/ProcessorConfiguration.java:45]: same as above

File: src/main/java/org/craftercms/studio/api/v1/asset/processing/ProcessorPipelineConfiguration.java (hunk 3 above)
Comment (dejan-brkic) [reply src/main/java/org/craftercms/studio/api/v1/asset/processing/ProcessorPipelineConfiguration.java:31]: same as above

File: src/main/java/org/craftercms/studio/api/v1/aws/elastictranscoder/TranscoderJob.java (hunk 4 above)
Comment (dejan-brkic) [reply src/main/java/org/craftercms/studio/api/v1/aws/elastictranscoder/TranscoderJob.java:30]: same as above

File: src/main/java/org/craftercms/studio/api/v1/aws/elastictranscoder/TranscoderOutput.java (hunk 5 above)
Comment (dejan-brkic) [reply src/main/java/org/craftercms/studio/api/v1/aws/elastictranscoder/TranscoderOutput.java:37]: same as above

File: src/main/java/org/craftercms/studio/api/v1/aws/elastictranscoder/TranscoderProfile.java (hunk 6 above)
Comment (dejan-brkic) [reply src/main/java/org/craftercms/studio/api/v1/aws/elastictranscoder/TranscoderProfile.java:32]: same as above

**Review by sumerjabri — APPROVED** [review]

**Pull request merged**